service/src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -394,15 +397,27 @@ private FlightMap getResultMap(FlightState flightState) {
     return resultMap;
   }
 
+  /**
+   * Ensure the user in the user request has permission to read the workspace associated with the
+   * Job ID. Throws ForbiddenException if not. If the workspace does not exist, treat this as
+   * the user having access (i.e. skip validateWorkspaceAndAction check).
+   *
+   * @param jobId - ID of running job
+   * @param userRequest - original user request
+   */
   private void verifyUserAccess(String jobId, AuthenticatedUserRequest userRequest) {
     try {
       FlightState flightState = stairwayComponent.get().getFlightState(jobId);
       FlightMap inputParameters = flightState.getInputParameters();
-      String flightSubjectId =
-          inputParameters.get(JobMapKeys.SUBJECT_ID.getKeyName(), String.class);
-      if (!StringUtils.equals(flightSubjectId, userRequest.getSubjectId())) {
-        throw new JobUnauthorizedException("Unauthorized");
-      }
+      UUID workspaceId = inputParameters.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+
+      // If the workspace doesn't exist, don't try to validate it
+      final WorkspaceService workspaceService = flightBeanBag.getWorkspaceService();
+      final WorkspaceDao workspaceDao = flightBeanBag.getWorkspaceDao();
+
+      workspaceDao.getWorkspaceIfExists(workspaceId).ifPresent(w ->
```

Comment:
This is not ideal, since someone can probe completed flights of a deleted workspace.
Is the only case we need to allow the delete workspace case? Unfortunately, there is no information in the FlightState to tell us what the flight class is.

Anyway, I do not have a  better idea for fix right now.

Reply:
Delete workspace is the primary offender, and it caused tons of test failures. I'm not sure if it's the only one.

Reply:
You're saying this change allows probing that would otherwise not be permitted?

Reply:
Yes. It is a bit obscure. Imagine that I know the UUID of a workspace and the flight ids of some flights that interest me. I am not a member of the workspace. The workspace gets deleted. I can now get the parameters of those flights when previously, I couldn't see them.

Since we often serialize the auth token, there is a window between workspace deletion and token timeout where I could swipe the token and act as the job submitter.

The best defense would be to either stop or delay deleting the Sam permission information. If it existed until all traces of the workspace were gone from the Stairway database, then it would be safe to delete.

How about I make a ticket for this and we go ahead with the change as a short term fix.

Reply:
I made https://broadworkbench.atlassian.net/browse/PF-1313

Reply:
You are right. WSM does not return the parameter information. So this is not so bad.